### ISSUE TITLE
[FEAT] Expose struct field decorator API

### ISSUE THREAD

**Issue by JaimeGensler** [JaimeGensler/thyseus#4]
It's currently not really possible to create custom struct field decorators. The current internal API isn't too bad, but should probably be cleaned up / rethought before being exposed.

Blocked by https://github.com/JaimeGensler/thyseus/issues/3

**Comment by JaimeGensler** [comment]
On hold while investigating serializing/deserializing
